### ISSUE TITLE
When using useDocument or useCollection the render method gets called multiple times

### ISSUE THREAD

**Issue by riebel** [nandorojo/swr-firestore#92]
When I use the useDocument or useCollection hooks, my component gets rerendered about five times. 
Is there a way to prevent that or am I maybe just stupid and missing something ;) ?

```
export default function DocComponent() {
  const { data, loading } = useDocument('path/to/collection/doc')

  if (!data || loading) return null

  console.log('render')

  return (
    <View>
      <Text>XYZ</Text>
    </View>
  )
}
```

the console outputs:

```
render
render
render
render
render
```

**Comment by nandorojo** [comment]
Interesting. Maybe make a code sandbox?

**Comment by riebel** [comment]
I created a code sandbox and was not able to reproduce the mentioned behavior. Must be a bug on my side, Thanks for your time.

**Comment by riebel** [comment]
As I was testing again, the issue seems to be a bug in combination with Expo Fast Refresh.

The component gets rerendered multiple times by Expo if using Fast refresh.
At the initial load everything is fine and loads only once, but if you then change anything and Fast Refresh kicks in, the component gets rerendered multiple times. 
If you remove useCollection/useDocument, the component only renders once when changed and Fast Refresh is enabled.

As I was not able to generate a working Expo Snack, I have created a simple [Repo](https://github.com/riebel/swr-firestore-test) for testing. 
Just clone, install dependencies with yarn and start with expo start.
I only tested the bug on Android emulator yet.

[https://github.com/riebel/swr-firestore-test](https://github.com/riebel/swr-firestore-test)
